### ISSUE TITLE
TypeError: Response body object should not be disturbed or locked

### ISSUE THREAD

**Issue by adamlong5** [gptlabs/openai-streams#17]
Thanks for this awesome library! 

It's been working well, but today I started seeing the following error when I `return new Response(stream)` from a Next 13 route handler:
```
TypeError: Response body object should not be disturbed or locked
```

It will always return successfully the first time, but on any subsequent attempt to invoke the route, I get the error about the response body.

I'm unfortunately not very experienced with returning a ReadableStream via a Next, so I was hoping someone might see an obvious flaw in my Next 13 Beta (`/app` directory) route handler. However, I'm not doing anything different than your docs show:
```typescript
import { type NextRequest } from 'next/server'
import { OpenAI } from "openai-streams";

export const runtime = 'experimental-edge'; // Run on Edge Functions

export async function POST(request: NextRequest) {
  try {
    const stream = await OpenAI(...)
    return new Response(stream)
  } catch (error) {
    console.error(error)
  }
}
```

I'm invoking this route handler on the front-end by using the `useTextBuffer` hook from your `nextjs-openai` library.

**Comment by ctjlewis** [comment]
Thanks for opening this. No, your pattern looks correct, this seems to be our bug.

Upgrading `nextjs-openai` and `openai-streams` does not resolve the issue?

**Comment by ctjlewis** [comment]
We'll get this figured out, but I might need to press you for a reproduction (either a repro or a Replit); the demo uses the standard text buffer and StreamingText but seems to have no issues:

https://github.com/gptlabs/nextjs-openai/blob/master/src/pages/index.tsx

https://nextjs-openai.vercel.app

**Comment by adamlong5** [comment]
Ah, figured it out. Somehow I was on an older version (though I installed recently). It was returning the raw JSON deltas from OpenAI, so I had a custom `ContentFilterStream` that I was piping your stream through to get just the text content out of the JSON as each token streams in. Apparently, that was causing the issue because the filter was persisting across API calls.

Thanks for rubber ducking and thanks again for the library. Also your twitter is 🔥 . Closing this one.
